build.gradle

```diff
@@ -0,0 +1,152 @@
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+buildscript {
+    apply from: 'ext.gradle'
+
+    ext {
+        spineRepository = 'http://maven.teamdev.com/repository/spine'
+        spineSnapshotsRepository = 'http://maven.teamdev.com/repository/spine-snapshots'
+    }
+
+    repositories {
+        jcenter()
+
+        maven { url = spineRepository }
+        maven { url = spineSnapshotsRepository }
+
+        mavenCentral()
+        mavenLocal()
+    }
+
+    dependencies {
+        classpath group: 'com.google.guava', name: 'guava', version: guavaVersion
+        classpath(group: 'com.google.protobuf', name: 'protobuf-gradle-plugin', version: protobufGradlePluginVersion) {
+            // exclude an old Guava version
+            exclude group: 'com.google.guava'
+        }
+        classpath "org.junit.platform:junit-platform-gradle-plugin:${jUnitPlatformVersion}"
+        classpath "io.spine.tools:spine-model-compiler:${spineBaseVersion}"
+    }
+}
+
+allprojects {
+    apply from: "$rootDir/ext.gradle"
+    apply plugin: 'java'
+    apply plugin: 'jacoco'
+    apply plugin: 'idea'
+
+    repositories {
+        jcenter()
+
+        maven { url = spineRepository }
+        maven { url = spineSnapshotsRepository }
+
+        mavenCentral()
+        mavenLocal()
+    }
+
+    group = 'io.spine.server'
+    version = spineVersion
+}
+
+subprojects {
+    project.ext {
+        sourcesRootDir = "${projectDir}/src"
+        generatedRootDir = "${projectDir}/generated"
+
+        generatedJavaDir = "${generatedRootDir}/main/java"
+        generatedTestJavaDir = "${generatedRootDir}/test/java"
+
+        generatedGrpcDir = "${generatedRootDir}/main/grpc"
+        generatedTestGrpcDir = "${generatedRootDir}/test/grpc"
+
+        generatedSpineDir = "${generatedRootDir}/main/spine"
+        generatedTestSpineDir = "${generatedRootDir}/test/spine"
+    }
+
+    apply plugin: 'com.google.protobuf'
+    apply plugin: 'org.junit.platform.gradle.plugin'
+    apply plugin: 'io.spine.tools.spine-model-compiler'
+
+    sourceCompatibility = JavaVersion.VERSION_1_7
+    targetCompatibility = JavaVersion.VERSION_1_7
+
+    dependencies {
+        compile (group: 'io.spine', name: 'spine-testutil-server', version: spineVersion)
+        compile group: 'org.slf4j', name: 'slf4j-api', version: slf4jVersion
+        compile group: 'com.google.code.findbugs', name: 'jsr305', version: '3.0.0'
+        compile group: 'com.google.guava', name: 'guava', version: guavaVersion
+
+        testCompile group: 'org.hamcrest', name: 'hamcrest-all', version: '1.3'
+        testCompile("io.spine:spine-testutil-base:${spineBaseVersion}") {
+            // We are using JUnit 5.x instead of 4.x dependency of Spine testing framework.
```

Comment:
Are we using JUnit5 with JDK7? I don't think so.
  
Same below. For several more times.